### PR TITLE
Support resource name

### PR DESCRIPTION
**Pull request by koxudaxi**
The PR supports the resource name to connect the database.
## Related issues
https://github.com/koxudaxi/py-data-api/issues/31

### CODE REVIEW

File: pydataapi/pydataapi.py [hunk 1]
```diff
@@ -356,8 +366,12 @@ def __init__(
         transaction_id: Optional[str] = None,
         client: Optional[boto3.session.Session.client] = None,
         rollback_exception: Optional[Type[Exception]] = None,
+        rds_client: Optional[boto3.session.Session.client] = None,
     ) -> None:
-        self.resource_arn: str = resource_arn
+        if resource_arn.startswith('arn:aws:rds:'):
+            self.resource_arn: str = resource_arn
+        else:
+            self.resource_arn = find_arn_by_resource_name(resource_arn, rds_client)
```
Comment (koxudaxi) [inline pydataapi/pydataapi.py]: We set resource name to `resource_arn`,
However, I should think a better way 🤔

File: pydataapi/pydataapi.py [hunk 2]
```diff
@@ -356,8 +366,12 @@ def __init__(
         transaction_id: Optional[str] = None,
         client: Optional[boto3.session.Session.client] = None,
         rollback_exception: Optional[Type[Exception]] = None,
+        rds_client: Optional[boto3.session.Session.client] = None,
```
Comment (koxudaxi) [inline pydataapi/pydataapi.py]: Should we accept access_key_id, secret_access_key, region and endpoint?

**Comment by koxudaxi** [comment]
@Rubyj 
Would you please review PR?

File: pydataapi/pydataapi.py (hunk 1 above)
Comment (Rubyj) [reply pydataapi/pydataapi.py]: I think we can either pass `resource_arn` or `resource_name` to `connect_args`. If we pass `resource_arn` then we do nothing. If we pass `resource_name` then we retireve arn.

File: pydataapi/pydataapi.py (hunk 2 above)
Comment (Rubyj) [reply pydataapi/pydataapi.py]: Hm. We could. Although, that might be a bigger change. I think it is safe to assume most systimes have that information in `~/.aws`

File: pydataapi/pydataapi.py [hunk 3]
```diff
@@ -347,16 +347,38 @@ def build(self) -> Dict[str, Any]:
...
         database: Optional[str] = None,
         transaction_id: Optional[str] = None,
         client: Optional[boto3.session.Session.client] = None,
         rollback_exception: Optional[Type[Exception]] = None,
+        rds_client: Optional[boto3.session.Session.client] = None,
     ) -> None:
+        if resource_name:
+            if resource_arn:
+                raise DataAPIError(
+                    f'resource_name should be set without resource_arn. resource_arn: {resource_arn},'
```
Comment (Rubyj) [inline pydataapi/pydataapi.py]: If they are both set is it really a problem? If both are set maybe we can just use arn?

Comment (koxudaxi) [reply pydataapi/pydataapi.py]: I think implicit behavior is not good for users.
Do you think about it?
documents may be cover the problem.

File: pydataapi/pydataapi.py (hunk 1 above)
Comment (koxudaxi) [reply pydataapi/pydataapi.py]: I just changed it.

File: pydataapi/pydataapi.py (hunk 2 above)
Comment (koxudaxi) [reply pydataapi/pydataapi.py]: OK, I don't touch this argument. Users may want to set a client object.

File: pydataapi/pydataapi.py (hunk 3 above)
Comment (Rubyj) [reply pydataapi/pydataapi.py]: That is a good point. Ok we can force users to set only one or other if you would like.

**Comment by koxudaxi** [comment]
@Rubyj 
Thank you for reviewing the PR 🎉

**Pull request merged**